model/task/task.go

```diff
@@ -4037,3 +4039,100 @@ func (t *Task) FindAbortingAndResettingDependencies() ([]Task, error) {
+// with accompanying breakdowns of priority and rank value.
+type SortingValueBreakdown struct {
+	TaskGroupLength    int64
+	TotalValue         int64
+	PriorityBreakdown  PriorityBreakdown
+	RankValueBreakdown RankValueBreakdown
+}
+
+// PriorityBreakdown contains information on how much various factors impacted the custom
+// priority value that is used in the queue sorting value equation.
+type PriorityBreakdown struct {
+	// InitialPriorityImpact represents how much of the total priority can be attributed to the
+	// original priority set on a task by a user.
+	InitialPriorityImpact int64
+	// TaskGroupImpact represents how much of the total priority can be attributed to a
+	// task being in a task group.
+	TaskGroupImpact int64
+	// GenerateTaskImpact represents how much of the total priority can be attributed to a
+	// task having a generate.tasks command in it.
+	GenerateTaskImpact int64
+	// CommitQueueImpact represents how much of the total priority can be attributed to a task
+	// being in the commit queue.
+	CommitQueueImpact int64
+}
+
+// RankValueBreakdown contains information on how much various factors impacted the custom
+// rank value that is used in the queue sorting value equation.
+type RankValueBreakdown struct {
+	// CommitQueueImpact represents how much of the total rank value can be attributed to a task
+	// being in the commit queue.
+	CommitQueueImpact int64
+	// NumDependentsImpact represents how much of the total rank value can be attributed to the number
+	// of tasks that are dependents of a task.
+	NumDependentsImpact int64
+	// EstimatedRuntimeImpact represents how much of the total rank value can be attributed to the
+	// estimated runtime of a task.
+	EstimatedRuntimeImpact int64
+	// MainlineWaitTimeImpact represents how much of the total rank value can be attributed to
+	// how long a mainline task has been waiting for dispatch.
+	MainlineWaitTimeImpact int64
+	// StepbackImpact represents how much of the total rank value can be attributed to
+	// a task being activated by the stepback process.
+	StepbackImpact int64
+	// PatchImpact represents how much of the total rank value can be attributed to a task
+	// being part of a patch.
+	PatchImpact int64
+	// PatchWaitTimeImpact represents how much of the total rank value can be attributed to
+	// how long a patch task has been waiting for dispatch.
+	PatchWaitTimeImpact int64
+}
+
+const (
+	priorityBreakdownAttributePrefix = "evergreen.priority_breakdown"
+	rankBreakdownAttributePrefix     = "evergreen.rank_breakdown"
+	priorityScaledRankAttribute      = "evergreen.priority_scaled_rank"
+)
+
+// SetSortingValueBreakdown saves a full breakdown which compartmentalizes each factor that played a role in computing the
+// overall value used to sort it in the queue, and creates a honeycomb trace with this data to enable dashboards/analysis.
+func (t *Task) SetSortingValueBreakdown(ctx context.Context, breakdown SortingValueBreakdown) {
```

Comment:
extreme nit -- could we call this "SetSortingValueBreakdownAttributes" ? without that, it's more in line with a function name that sets something in the DB, which this function doesn't do